packages/notifier/src/notifier.js

```diff
@@ -1,32 +1,31 @@
 /// <reference types="ses"/>
 
 import { assert } from '@agoric/assert';
-import { makePromiseKit } from '@endo/promise-kit';
-import { E } from '@endo/eventual-send';
-import { Far } from '@endo/marshal';
-import { makeAsyncIterableFromNotifier } from './asyncIterableAdaptor.js';
+import { E, Far } from '@endo/far';
 
 import './types-ambient.js';
+import { makePublishKit } from './publish-kit.js';
+import { subscribeLatest } from '../tools/subscribe.js';
 
 /**
  * @template T
  * @param {ERef<BaseNotifier<T> | NotifierInternals<T>>} sharableInternalsP
- * @returns {AsyncIterable<T> & SharableNotifier<T>}
+ * @returns {Notifier<T>}
  */
 export const makeNotifier = sharableInternalsP => {
-  const asyncIterable = makeAsyncIterableFromNotifier(sharableInternalsP);
-
-  /** @type {AsyncIterable<T> & SharableNotifier<T>} */
+  /** @type {Notifier<T>} */
   const notifier = Far('notifier', {
-    ...asyncIterable,
+    ...subscribeLatest(sharableInternalsP),
+    getUpdateSince: async updateCount =>
+      E(sharableInternalsP).getUpdateSince(updateCount),
 
     /**
      * Use this to distribute a Notifier efficiently over the network,
      * by obtaining this from the Notifier to be replicated, and applying
      * `makeNotifier` to it at the new site to get an equivalent local
      * Notifier at that site.
      */
-    getSharableNotifierInternals: () => sharableInternalsP,
+    getSharableNotifierInternals: async () => sharableInternalsP,
     getStoreKey: () => harden({ notifier }),
```

Comment:
```suggestion
    /** @deprecated */
    getStoreKey: () => harden({ notifier }),
```
(it's also worth noting that getStoreKey is not part of the Notifier interface; it is part of StoredFacet (which also requires other methods that are not added here).